server/notification-providers/smtp.js

```diff
@@ -36,76 +37,80 @@ class SMTP extends NotificationProvider {
+            const engine = new Liquid();
             if (customSubject !== "") {
-
-                // Replace "MACROS" with corresponding variable
-                let replaceName = new RegExp("{{NAME}}", "g");
-                let replaceHostnameOrURL = new RegExp("{{HOSTNAME_OR_URL}}", "g");
-                let replaceStatus = new RegExp("{{STATUS}}", "g");
-
-                // Lets start with dummy values to simplify code
-                let monitorName = "Test";
-                let monitorHostnameOrURL = "testing.hostname";
-                let serviceStatus = "⚠️ Test";
-
-                if (monitorJSON !== null) {
-                    monitorName = monitorJSON["name"];
-
-                    if (monitorJSON["type"] === "http" || monitorJSON["type"] === "keyword" || monitorJSON["type"] === "json-query") {
-                        monitorHostnameOrURL = monitorJSON["url"];
-                    } else {
-                        monitorHostnameOrURL = monitorJSON["hostname"];
-                    }
-                }
-
-                if (heartbeatJSON !== null) {
-                    serviceStatus = (heartbeatJSON["status"] === DOWN) ? "🔴 Down" : "✅ Up";
-                }
-
-                // Break replace to one by line for better readability
-                customSubject = customSubject.replace(replaceStatus, serviceStatus);
-                customSubject = customSubject.replace(replaceName, monitorName);
-                customSubject = customSubject.replace(replaceHostnameOrURL, monitorHostnameOrURL);
-
-                subject = customSubject;
+                const tpl = engine.parse(customSubject);
+                subject = await engine.render(tpl, context);
+            }
+            if (customBody !== "") {
+                const tpl = engine.parse(customBody);
+                body = await engine.render(tpl, context);
             }
-        }
-
-        let transporter = nodemailer.createTransport(config);
-
-        let bodyTextContent = msg;
-        if (heartbeatJSON) {
-            bodyTextContent = `${msg}\nTime (${heartbeatJSON["timezone"]}): ${heartbeatJSON["localDateTime"]}`;
         }
 
         // send mail with defined transport object
+        let transporter = nodemailer.createTransport(config);
         await transporter.sendMail({
             from: notification.smtpFrom,
             cc: notification.smtpCC,
             bcc: notification.smtpBCC,
             to: notification.smtpTo,
             subject: subject,
-            text: bodyTextContent,
+            text: body,
         });
 
         return "Sent Successfully.";
     }
+
+    /**
+     * Generate context for LiquidJS
+     * @param msg {string}
+     * @param monitorJSON {Object|null}
+     * @param heartbeatJSON {Object|null}
+     * @returns {{STATUS: string, HOSTNAME_OR_URL: string, NAME: string, monitorJSON: Object|null, heartbeatJSON: Object|null, msg: string}}
```

Comment:
Could you fix this JSDoc comment please? It will cause errors/warnings from eslint in V2 currently. The main things are param and returns need comments and object is preferred.

Reply:
I will, once https://github.com/louislam/uptime-kuma/pull/2720 is merged into master and rebasing is less of a pain of having to track two upstreams